public/app/plugins/datasource/cloudwatch/datasource.js

```diff
@@ -191,7 +185,20 @@ function (angular, _) {
           });
         }
 
-        return this.getDimensionValues(region, namespace, metricName, dimensions);
+        return this.getDimensionValues(region, namespace, metricName, dimensions).then(function(result) {
+          return _.map(result, function(dimensions) {
+            var values = _.chain(dimensions)
+            .sortBy(function(dimension) {
+              return dimension.Name;
+            })
+            .map(function(dimension) {
+              return dimension.Name + '=' + dimension.Value;
```

Comment:
this seems very wrong, why return this? getDimensionValues should return values for the current selected dimension, so getDimensionValues should take a dimension key as parameter. And filtering should be here not in query_ctrl